wpninjas/ninja-forms
# Adds Basic Error Logging

This PR utilises the "logger" system already in Ninja Forms core to add basic error logs, which are then displayed upon the system status page. This allows for support techs to see system errors without the need to load a user's site, which should save them significant time.

Currently, these are the errors captured:

- JS Errors on the front-end.
- Submission errors.
- License activation errors.

This PR also changes the way that users copy their system status information. Previously, the user clicked a button to show a textarea and then copied the text to their clipboard. This PR adds a "Copy System Status to Clipboard" button to the system status page.

How to test:

- Add a plugin or modify some JS code to get a JS error upon form display.
- Throw a PHP error upon form submission.
- Try to activate an invalid license for an add-on.

All of those actions should throw errors that are captured and displayed on the system status page. If you click the "Copy to Clipboard" button, you should be able to paste the system status elsewhere.

TODO:

This is a first-pass at error reporting for support purposes; I fully expect that this will need to be modified and evolve as support finds new ways to use it.

@wpnzach I need you to take a look at the system status page and make suggestions about wording and layout for the "Copy to Clipboard" button, as well as any instructions that you'd like the user to see in that location.
